djconnectwise/sync.py

```diff
@@ -1235,12 +1235,28 @@ def create_new_entry(self, target, **kwargs):
             target_data['id'] = target.id
             target_data['type'] = target.record_type
         else:
-            raise InvalidObjectException("Invalid target type for TimeEntry \
-creation: " + str(target.__class__) + ".")
+            raise InvalidObjectException(
+                "Invalid target type for TimeEntry "
+                "creation: " + str(target.__class__) + "."
+            )
         time_client = api.TimeAPIClient()
         instance = time_client.post_time_entry(target_data, **kwargs)
         return self.update_or_create_instance(instance)
 
+    def callback_sync(self, filter_params):
+        sync_job_qset = self.get_sync_job_qset()
+
+        if sync_job_qset.exists():
+            last_sync_job_time = sync_job_qset.last().start_time.isoformat()
+            self.api_conditions.append(
+                "lastUpdated>[{0}]".format(last_sync_job_time)
+            )
+        results = SyncResults()
+        results = self.get(results, )
+
+        return results.created_count, results.updated_count, \
```

Comment:
This is still running a full sync when it should be running a partial sync now.

Reply:
I'm not sure I understand. It should be running a full sync (except no pruning) under some circumstances.

As it is now it will sync all time entries on the ticket but only if there is **not** a prior successful sync job. If there is a successful time entry sync job then it should add the api condition and only return time entries that were created after the last time entry sync job.  

It took me awhile to wrap my head around some parts so I could definitely be missing something. Let me know if I'm completely missing something

Reply:
Well the issue name is "In ticket callback, perform **partial** time entry sync", and in the issue description it says: "Done when in the callback handler we do a partial sync instead", so I just assumed it was supposed to a partial sync :laughing: 

Let me know if I missed a conversation or something that changed the plan

Reply:
Yeah you are right. I shouldn't have said if should running a full sync under some circumstances. That's just confusing. 

 I should've said that if no time entry sync jobs exists before the callback is executed then the `callback_sync` will sync all time entries on the ticket. That should be the only case that all time entries on the ticket get synced. 

Are you seeing it sync all time entries on the ticket every time `callback_sync` runs?

The description mentions that 
> only fetching those time entries for the tickets that have been updated in CW since our last recorded successful time entry sync job

Another thing to note is that callback syncs don't actually generate sync job logs. So if you create a time entry and callback sync runs and then you create another time entry that also runs callback sync it will sync both time entries not just the most recently created one.

Reply:
Ok, I see where the confusion arose here. I was abstracting what defined a "full" sync too much. I thought this was still somehow being called as a full sync but it is not, this is where it would be designed as a "full" or "partial" sync. Seeing the deleted count in the logs threw me off. Then I got really confused by `running a full sync (except no pruning)`, because the pruning is what makes a full sync a full sync.

I see now that `sync_children` is calling `callback_sync`, which is formatting the message and expecting a deleted count, which is why you provided it. In a normal partial sync, the deleted count isn't even displayed, just created and updated. So this is ACTUALLY a partial sync, but in the logs it looks like a full sync because the deleted count is still being shown.

I don't think it is worth refactoring `sync_children` just so we don't show a deleted count in this one case, so instead put a comment in `TimeEntrySynchronizer.callback_sync` that mentions this is a partial sync and even though we are displaying a deleted count it will never actually delete anything.

Reply:
Thanks for clarifying. Sounds good. Sorry for creating some confusion here.  I've added a comment now.